src/composer/src/build.rs

```diff
@@ -421,56 +430,77 @@ impl BuildState for DefaultBuilder {
         let p = state.get_param_id(&id);
         let output_path = self.comp_obj_path(&id, &state)?;
 
-        let c = component(&state, &id);
-        //rebuild process starts
+        let comp_log = self.comp_file_path(&id, &"compilation.log".to_string(), &state)?;
+
         let header_file_path =
             self.comp_file_path(&id, &"component_constants.h".to_string(), &state)?;
-        self.comp_const_header_file(&header_file_path, &id, &state)?;
 
-        let dep_cmd = comp_gen_make_cmd(
-            &output_path,
-            p.param_prog(),
-            p.param_fs(),
-            &header_file_path,
-            CmdOpts::DEPINFO,
-            &id,
-            &state,
-        );
-        let (out1, err1) = exec_pipeline(vec![dep_cmd.clone()]);
+        self.comp_const_header_file(&header_file_path, &id, &state)?;
 
-        let rebuild_cmd = format!(
-            r#"make -C src REBUILD_DIRS="{}" COMP_CONST_H="-include {}" component_rebuild"#,
-            out1, header_file_path
-        );
-        let (out2, err2) = exec_pipeline(vec![rebuild_cmd.clone()]);
+        //rebuild process starts
+        if let Some(flag) = &self.rebuildflag {
+            if flag == "REBUILD" {
+                let dep_cmd = comp_gen_make_cmd(
+                    &output_path,
+                    p.param_prog(),
+                    p.param_fs(),
+                    &Some(header_file_path.clone()),
+                    CmdOpts::DEPINFO,
+                    &id,
+                    &state,
+                );
+                let (out1, err1) = exec_pipeline(vec![dep_cmd.clone()]);
+
+                let rebuild_cmd = format!(
+                    r#"make -C src REBUILD_DIRS="{}" COMP_CONST_H="-include {}" component_rebuild"#,
+                    out1, header_file_path
+                );
+                let (out2, err2) = exec_pipeline(vec![rebuild_cmd.clone()]);
+
+                emit_file(
+                    &comp_log,
+                    format!(
+                        "Dep Command: {}\nCompilation output:{}\ncompilation errors:{}\n
```

Comment:
I have no idea what the user will think when they see `Dep Command`. Can you use a string that people might understand, like "Command to determine a component's dependencies:"...?

Reply:
updated the log message